migrate/live-migration/live-migration-from-postgres.md

```diff
@@ -0,0 +1,321 @@
+3][step-3]). The point that marks the beginning of the replication and
+buffering of changes is given by the exported snapshot. The larger the
+database, the more time it takes to perform the initial migration, and the
+longer the buffered files need to be stored.
+
+## 3. Migrate roles and schema from source to target
+
+### 3a. Dump the database roles from the source database
+
+<DumpDatabaseRoles />
+
+### 3b. Dump the database schema from the source database
+
+```sh
+pg_dump -d "$SOURCE" \
+  --format=plain \
+  --quote-all-identifiers \
+  --no-tablespaces \
+  --no-owner \
+  --no-privileges \
+  --schema-only \
+  --file=dump.sql \
+  --snapshot=$(cat /tmp/pgcopydb/snapshot)
+```
+
+- `--schema-only` is used to dump only the object definitions (schema), not
+  data.
+
+- `--snapshot` is use to specified the synchronized [snapshot][snapshot] when
+  making a dump of the database.
+
+<ExplainPgDumpFlags />
+
+### 3c. Load the roles and schema into the target database
+
+```sh
+psql -X -d "$TARGET" \
+  -v ON_ERROR_STOP=1 \
+  --echo-errors \
+  -f roles.sql \
+  -f dump.sql
+```
+
+## 4. Convert hypertables and enable Timescale features
+
+This is the ideal point to convert regular tables into hypertables. In simple
+terms, you'll want to convert the tables that contain time series data. For
+each table that's going to be converted to a Hypertable in the target database,
+run the following command:
+
+```sh
+psql -X -d "$TARGET" \
+  -v ON_ERROR_STOP=1 \
+  -c "SELECT create_hypertable('<table name>', '<time column name>')"
+```
+
+A more detailed explanation can be found in the [hypertable
+documentation][hypertable-docs].
+
+Once the table is converted, you can follow the guides to enable more Timescale
+features like [retention][retention-docs], [compression][compression-docs], and
+[data tiering][data-tiering-docs].
+
+This step is optional. Although it's strongly recommended to perform it at this
+point, it can be done after all the data has been migrated, since Hypertables
+can be created from tables with existing data.
+
+The disadvantage of converting after inserting all the data is that the table
+must be locked while the data is being partitioned. This means that no new data
+can be written into the table while the table is being converted. The larger
+the table, the longer it is going to remain locked. Conversely, inserting into
+an already converted Hypertable is a fast operation that partitions the data
+while it's being inserted.
+
+## 5. Migrate initial data from source to target
+
+Use `pgcopydb` and the snapshot ID obtained in the [step 2][step-2]
+(`/tmp/pgcopydb/snapshot`) to copy the data:
+
+```sh
+pgcopydb copy table-data \
+  --source "$SOURCE" \
+  --target "$TARGET" \
+  --snapshot $(cat /tmp/pgcopydb/snapshot) \
+  --split-tables-larger-than <same-table concurrency size threshold>  \
+  --table-jobs <jobs: usually the number of core in SOURCE>
+```
+
```

Comment:
This command runs for (presumably) a long time. Should we give users a warning about that? And tell them to wait for this command to complete before proceeding to the next step?